src/traces/ohlc/calc.js

```diff
@@ -0,0 +1,163 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Lib = require('../../lib');
+var _ = Lib._;
+var Axes = require('../../plots/cartesian/axes');
+
+function calc(gd, trace) {
+    var xa = Axes.getFromId(gd, trace.xaxis);
+    var ya = Axes.getFromId(gd, trace.yaxis);
+
+    var tickLen = convertTickWidth(gd, xa, trace);
+    var minDiff = trace._minDiff;
+    delete trace._minDiff;
+    var x = trace._xcalc;
+    delete trace._xcalc;
```

Comment:
Why do we need to delete `_xcalc` and `_minDiff`?

Reply:
per https://github.com/plotly/plotly.js/pull/2561/files/82677ac8745ca56da740302977533dffcd5521b6#diff-6877a6db77ade9f43ac4ccd6acbb2713R114:

> when a trace uses this in its calc step it deletes _minDiff, so that next calc this is done again in case the data changed.

I guess in principle we wouldn't need to delete `_xcalc` but it makes me feel better, as we're guaranteed it doesn't get used outside of `calc`. It's only there for 🐎 in the first place, so we don't have to calculate it in both `convertTickWidth` (for all traces) and then again in `calc` (for each trace separately)

Reply:
Thanks for the :books: 

So in theory we could set `trace._minDiff = null` instead of using `delete` here with no side-effects? Some benchmarks: https://jsperf.com/delete-vs-undefined-vs-null/16

Reply:
That's true - sure, I'll change it to `= null`.

Reply:
-> null https://github.com/plotly/plotly.js/pull/2561/commits/f498bd04e0541a4b6de7f7785d752fd6f1bb6481